extension-push/src/java/com/defold/push/Push.java

```diff
@@ -177,6 +179,7 @@ public void run() {
                     startFirebase(activity);
                     loadSavedLocalMessages(activity);
                     loadSavedMessages(activity);
+                    flushStoredNotifications();
```

Comment:
The problem is that if a game calls `push.register()` and  `push.set_listener(cb)` at the same time (which is very likely) then `push.register()` will finish after `push.set_listener()`.

This is a problem since any saved notifications haven't been loaded and the listener will not trigger. Doing a flush of saved notifications immediately after push.register() will solve this problem.

Reply:
Not sure if anyone will ever (re)move this line, but that sound like a good code comment to me :)